mesh/nurbs.hpp

```diff
@@ -282,6 +282,16 @@ class NURBSExtension
    friend class NURBSPatchMap;
 
 protected:
+   enum Mode
+   {
+      HP,
```

Comment:
`enum class` is preferable. Documentation for the types would be helpful (HP has multiple meanings).

Reply:
Add the `class` and changed `HP` to `H_1`. I'll also will add some Documentation.